### PR TITLE
fix: improve the composer.json file

### PR DESCRIPTION
**Pull request by dunglas**
Add missing PHP version requirement and a dev dependency.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -12,6 +12,7 @@
         }
     ],
     "require": {
+        "php": ">=7.4",
```
Comment (rennokki) [inline composer.json:15]: `illuminate/support` already has `php` constraint.

File: composer.json [hunk 2]
```diff
@@ -34,6 +35,7 @@
         "test": "vendor/bin/phpunit"
     },
     "require-dev": {
+        "ext-yaml": "*",
```
Comment (rennokki) [inline composer.json:38]: I'd put `ext-yaml` to [`suggest`](https://getcomposer.org/doc/04-schema.md#suggest), it is not required to use the entire package, only to load YAML files.

File: composer.json (hunk 1 above)
Comment (dunglas) [reply composer.json:15]: Sure, as most packages, but relying on transient dependencies isn't very reliable.

File: composer.json (hunk 2 above)
Comment (dunglas) [reply composer.json:38]: Yes but it is required to run the tests, so it must be in the `require-dev` section.

Comment (rennokki) [reply composer.json:38]: This means Composer won't alert if the user installs the package with `--no-dev`, so we might have an issue here.

**Pull request closed without merging**